core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/BlockCacheManager.java

```diff
@@ -0,0 +1,121 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.file.blockfile.cache;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.accumulo.core.conf.AccumuloConfiguration;
+import org.apache.accumulo.core.conf.Property;
+import org.apache.accumulo.start.classloader.vfs.AccumuloVFSClassLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class BlockCacheManager {
+
+  public static final String CACHE_PROPERTY_BASE = Property.GENERAL_ARBITRARY_PROP_PREFIX + "cache.block.";
+
+  private static final Logger LOG = LoggerFactory.getLogger(BlockCacheManager.class);
+  private static BlockCacheManager manager = null;
+
+  protected final Map<CacheType,BlockCache> caches = new HashMap<>();
+
+  /**
+   * Initialize the caches for each CacheType based on the configuration
+   *
+   * @param conf
+   *          accumulo configuration
+   */
+  public void start(AccumuloConfiguration conf) {
+    for (CacheType type : CacheType.values()) {
+      BlockCache cache = this.createCache(conf, type);
+      this.caches.put(type, cache);
+    }
+  }
+
+  /**
+   * Stop caches and release resources
+   */
+  public void stop() {
+    this.caches.clear();
+    close();
+  }
+  
+  private static synchronized void close() {
+	  manager = null;
+  }
+
+  /**
+   * Get the block cache of the given type
+   *
+   * @param type
+   *          block cache type
+   * @return BlockCache or null if not enabled
+   */
+  public BlockCache getBlockCache(CacheType type) {
+    return caches.get(type);
+  }
+
+  /**
+   * Create a block cache using the supplied configuration
+   *
+   * @param conf
+   *          cache configuration
+   * @return configured block cache
+   */
+  protected abstract BlockCache createCache(AccumuloConfiguration conf, CacheType type);
+
+  /**
+   * Get the BlockCacheFactory specified by the property 'tserver.cache.factory.class' using the AccumuloVFSClassLoader
+   *
+   * @param conf
+   *          accumulo configuration
+   * @return block cache manager instance
+   * @throws Exception
+   *           error loading block cache manager implementation class
+   */
+  public static synchronized BlockCacheManager getInstance(AccumuloConfiguration conf) throws Exception {
+    if (null == manager) {
+      String impl = conf.get(Property.TSERV_CACHE_FACTORY_IMPL);
+      Class<? extends BlockCacheManager> clazz = AccumuloVFSClassLoader.loadClass(impl, BlockCacheManager.class);
+      manager = (BlockCacheManager) clazz.newInstance();
```

Comment:
I'm ok with this considering that this is only used in TabletServerResourceManager.